From: Duncan Eastoe <deastoe@vyatta.att-mail.com>
Date: Tue, 13 Nov 2018 16:49:02 +0000
Subject: zebra: FPM netlink RTM_DELROUTE set RTN_UNICAST

Always initialise rtm_type to RTN_UNICAST to ensure rtm_type is not
RTN_UNSPEC when re == NULL and cmd == RTM_DELROUTE (which is always
the case today).
---
 zebra/zebra_fpm_netlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zebra/zebra_fpm_netlink.c b/zebra/zebra_fpm_netlink.c
index dcfd797..b6c38f4 100644
--- a/zebra/zebra_fpm_netlink.c
+++ b/zebra/zebra_fpm_netlink.c
@@ -331,6 +331,7 @@ static int netlink_route_info_fill(netlink_route_info_t *ri, int cmd,
 	ri->nlmsg_type = cmd;
 	ri->rtm_table = rib_table_info(rib_dest_table(dest))->table_id;
 	ri->rtm_protocol = RTPROT_UNSPEC;
+	ri->rtm_type = RTN_UNICAST;
 
 	/*
 	 * An RTM_DELROUTE need not be accompanied by any nexthops,
@@ -346,7 +347,6 @@ static int netlink_route_info_fill(netlink_route_info_t *ri, int cmd,
 	}
 
 	ri->rtm_protocol = netlink_proto_from_route_type(re->type);
-	ri->rtm_type = RTN_UNICAST;
 	ri->metric = &re->metric;
 
 	for (ALL_NEXTHOPS(re->ng, nexthop)) {
